### PR TITLE
logging: flowcontrol should block

### LINKED ISSUE #2714: Default FlowControlSettings LimitExceededBehavior
Hi I'm having an issue with my code, I recently started getting
com.google.api.gax.batching.FlowController$FlowControlRuntimeException: The maximum number of batch bytes: 10485760 have been reached.

I can see that's the default behaviour for LoggingOptions.getDefaultInstance().getService(); but the documentation isn't very clear on why its that way or what the alternatives are.
I can see from com.google.api.gax.batching.FlowControlSettings that there are 2 other alternatives Block and Ignore.

I think it would be helpful if somewhere in the documentation these options and their implications were explained and also how to switch between them.

I'd be interested to know if I can switch to Block and what that would mean for my program.

My program is pulling data from an activeMQ queue, logging it to google and then further processing it.
I have to restart the program or else the error continues on indefinitely. Could this error be caused by a message from the queue being above the limit? and if so if set to block will it process the message anyway?

How should this exception be handled so that I can log the data?

Thanks

### PR DESCRIPTION
**Pull request by pongad**
Fixes #2714.

### CODE REVIEW

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java [hunk 1]
```diff
@@ -138,6 +142,22 @@ public Void apply(UnaryCallSettings.Builder<?, ?> builder) {
...
+                  .setElementCountThreshold(1000L)
+                  .setRequestByteThreshold(1048576L)
+                  .setDelayThreshold(Duration.ofMillis(50))
+                  .setFlowControlSettings(
+                      FlowControlSettings.newBuilder()
+                          .setMaxOutstandingElementCount(100000L)
+                          .setMaxOutstandingRequestBytes(10485760L)
+                          .setLimitExceededBehavior(LimitExceededBehavior.Block)
+                          .build())
+                  .build());
```
Comment (garrettjonesgoogle) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java]: Is there no way to get the default batching settings and only change the LimiteExceededBehavior? This basically overrides whatever is in the GAPIC config for logging.

Comment (pongad) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java]: It's terrible but possible. Good point though.

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**